lineman/app/services/thread_query_service.coffee

```diff
@@ -52,15 +43,19 @@ angular.module('loomioApp').factory 'ThreadQueryService', (Records) ->
       parts = options.split ' '
       moment().startOf('day').subtract(parseInt(parts[0]), parts[1])
 
-    createFilter = (filter = 'show_all') ->
-      (viewData) ->
-        _.filter viewData, (thread) ->
-          return false if thread.isMuted() and filter != 'show_muted'
-          return false if thread.readerNotLoaded()
-          switch filter
-            when 'show_all'           then true
-            when 'show_muted'         then thread.isMuted()
-            when 'show_unread'        then thread.isUnread()
-            when 'show_participating' then thread.isParticipating()
-            when 'show_starred'       then thread.isStarred()
-            when 'show_proposals'     then thread.hasActiveProposal()
+    applyFilters = (view, filters, queryType) ->
```

Comment:
None of those (oh, except hasActiveProposal, now that I look at it, I can
do that one) are simple, I don't think, they're all reader methods.

It'd be interesting to see an alternate implementation of this where we're
displaying a list readers instead of a list of discussions, since most of
the functionality on dashboard / inbox is around 'what have YOU done with
this thread', ie, the reader.

On Thu, Aug 13, 2015 at 12:32 PM, Robert Guthrie <notifications@github.com>
wrote:

> In lineman/app/services/thread_query_service.coffee
> <https://github.com/loomio/loomio/pull/2369#discussion_r37015753>:
>
> > @@ -52,15 +43,19 @@ angular.module('loomioApp').factory 'ThreadQueryService', (Records) ->
> >        parts = options.split ' '
> >        moment().startOf('day').subtract(parseInt(parts[0]), parts[1])
> >
> > -    createFilter = (filter = 'show_all') ->
> > -      (viewData) ->
> > -        _.filter viewData, (thread) ->
> > -          return false if thread.isMuted() and filter != 'show_muted'
> > -          return false if thread.readerNotLoaded()
> > -          switch filter
> > -            when 'show_all'           then true
> > -            when 'show_muted'         then thread.isMuted()
> > -            when 'show_unread'        then thread.isUnread()
> > -            when 'show_participating' then thread.isParticipating()
> > -            when 'show_starred'       then thread.isStarred()
> > -            when 'show_proposals'     then thread.hasActiveProposal()
> > +    applyFilters = (view, filters, queryType) ->
>
> This looks really great.
>
> Have you considered using applyFind on the simple properties? I think
> using Find that means Loki can use indices rather than apply a function to
> every record.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r37015753>.
>